rust-toolchain.toml

```diff
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "nightly-2024-10-30"     # "1.82.0"
+channel = "nightly-2024-12-16"     # "1.82.0"
```

Comment:
Why is this necessary? I think in the interest of being able to potentially upstream these changes, I think we should avoid changing the Rust version if possible.